packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -412,7 +427,7 @@ abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecogni
       primaryPointer = event.pointer;
       initialPosition = event.position;
       if (deadline != null)
-        _timer = Timer(deadline, didExceedDeadline);
+        _timer = Timer(deadline, () => didExceedDeadlineWithEvent(event));
```

Comment:
This isn't ideal. If you have better ideas on how to do it, I'm all ears :)

Reply:
Ah, this is unfortunate. Though this event isn't entirely correct either right? Since you're sending something from the past and the past event also has a timeStamp. 

I haven't validated it but I suspect each onLongPressDown is preceded by a onTapDown or something. From the TextSelectionGestureDetector's perspective, it may be enough?

Reply:
Unfortunately, this is how the tap recognizer works. It doesn't send the tap down event until either a tap up occurs, or the deadline is exceeded. And in that case, the tap down is always from the past (even before my change). All my change does is it adds the `kind` field to the event.